### DISCUSSION TITLE
Will there be any replacement for pydantic.datetime_parse in V2?

### DISCUSSION THREAD

**Discussion by santigandolfo** [pydantic/pydantic#6204] [Question]
I see in the [migration docs](https://docs.pydantic.dev/dev-v2/migration/) that all the datetime_parse functions will be removed in v2. Is there any alternative?
Because for example im using the library odmantic and I see that they are using the function parse_datetime for datetime.
Cheers!

**Comment by adriangb** [comment]
Would something like this work?

```python
from pydantic import TypeAdapter

from datetime import datetime


parse_datetime = TypeAdapter(datetime).validate_json
```

If you wanted to accept either JSON or Python objects like the original `parse_datetime` you might need to wrap that in a function that does `isinstance(obj, datetime)` or something.

**Comment by wooparadog** [comment]
A quick glance at https://github.com/pydantic/pydantic/blob/abcf81ec104d2da70894ac0402ae11a7186c5e47/pydantic/datetime_parse.py.

I'd prefer just copy the old source file into my repo and just change the import.

**Comment by yashpct2807** [comment]
I am also using parse_datetime, Is there any alternative on pydantic v2?

**Reply by adriangb** [reply]
Is there any reason https://github.com/pydantic/pydantic/discussions/6204#discussioncomment-6266717 doesn't work for you?

**Reply by yashpct2807** [reply]
I create data type as below:
```
from datetime import datetime, timezone
from pydantic.datetime_parse import parse_datetime


class utc_datetime(datetime):
    @classmethod
    def __get_validators__(cls):
        yield parse_datetime
        yield cls.validate_tzinfo

    @classmethod
    def validate_tzinfo(cls, date_time):
        if date_time.tzinfo is None:
            return date_time.replace(tzinfo=timezone.utc)
        return date_time
```

same utc_datetime, I use in different pydantic schema.
